openmdao/core/tests/test_expl_comp.py

```diff
@@ -127,43 +124,533 @@ def test_compute_and_list(self):
+                                          shape=True,
+                                          bounds=True,
+                                          residuals=True,
+                                          scaling=True,
+                                          hierarchical=False,
+                                          print_arrays=False
+                                          )
+        self.assertEqual(outputs, [
+            ('p1.x', {'value': [ 12.], 'resids': [ 0.], 'units': 'inch', 'shape': (1,),
+                      'lower': [ 1.], 'upper': [ 100.], 'ref': 1.1, 'ref0': 2.1, 'res_ref': 1.1} ),
+            ('p2.y', {'value': [ 1.], 'resids': [ 0.], 'units': 'ft', 'shape': (1,),
+                      'lower': [ 2.], 'upper': [ 200.], 'ref': 1.2, 'ref0': 0.0, 'res_ref': 2.2}),
+            ('comp.z', {'value': [ 24.], 'resids': [ 0.], 'units': 'inch', 'shape': (1,),
+                        'lower': None, 'upper': None, 'ref': 1.0, 'ref0': 0.0, 'res_ref': 1.0} ),
+            ]
+            )
+
+        outputs = prob.model.list_outputs(
+                                          implicit=False,
+                                          values=True,
+                                          units=True,
+                                          shape=True,
+                                          bounds=True,
+                                          residuals=True,
+                                          scaling=True,
+                                          hierarchical=True,
+                                          print_arrays=False
+                                          )
+
+    def test_hierarchy_list_vars_options(self):
+
+        prob = Problem()
+        model = prob.model
+
+        model.add_subsystem('pz', IndepVarComp('z', np.array([5.0, 2.0])))
+
+        sub1 = model.add_subsystem('sub1', Group())
+        sub2 = sub1.add_subsystem('sub2', Group())
+        g1 = sub2.add_subsystem('g1', SubSellar())
+        g2 = model.add_subsystem('g2', SubSellar())
+
+        model.connect('pz.z', 'sub1.sub2.g1.z')
+        model.connect('sub1.sub2.g1.y2', 'g2.x')
+        model.connect('g2.y2', 'sub1.sub2.g1.x')
+
+        model.nonlinear_solver = NewtonSolver()
+        model.linear_solver = ScipyKrylov()
+        model.nonlinear_solver.options['solve_subsystems'] = True
+        model.nonlinear_solver.options['max_sub_solves'] = 0
+
+        g1.nonlinear_solver = NewtonSolver()
+        g1.linear_solver = LinearBlockGS()
+
+        g2.nonlinear_solver = NewtonSolver()
+        g2.linear_solver = ScipyKrylov()
+        g2.linear_solver.precon = LinearBlockGS()
+        g2.linear_solver.precon.options['maxiter'] = 2
+
+        prob.setup(check=False)
+        prob.run_driver()
+
+        # logging inputs
+        # out_stream - not hierarchical - extras - no print_arrays
+        stream = cStringIO()
+        prob.model.list_inputs(values=True,
+                                          units=True,
```

Comment:
Strange indenting

Reply:
will  fix. I wonder why all of the tests for proper formatting didn't flag those. It's usually so hard to make it happy!

Reply:
I can fix the column width issue. I actually did set the size of the first column based on the max width for that column since that width could vary a lot depending on the problem. I thought for the others it was easy to set an upper bound but I agree it can result in a lot of wasted space.